File: python/3/part1.py
import pathlib
import re

with open(pathlib.Path(__file__).parent.joinpath("input.txt"), "r", encoding="utf-8") as file:
    lines = file.readlines()

result = 0
pattern = r"(mul\((\d+),(\d+)\))"
for line in lines:
    line = line.rstrip("\n")
    numbs = re.findall(pattern, line)

    for list_n in numbs:
        result += int(list_n[1]) * int(list_n[2])

print(result)

File: c++/3/part1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <regex>

int main() {
    std::ifstream file("input.txt");
    if (!file.is_open()) {
        std::cerr << "Can not open the file." << std::endl;
        return 1;
    }
    // std::string input = "xmul(2,4)%&mul[3,7]!@^do_not_mul(5,5)+mul(32,64]then(mul(11,8)mul(8,5))";

    std::regex pattern(R"(mul\((\d+),(\d+)\))");
    std::smatch matches;

    int totalSum = 0;

    std::string input;

    while (std::getline(file, input)) {
        std::sregex_iterator currentMatch(input.begin(), input.end(), pattern);
        std::sregex_iterator end;

        while (currentMatch != end) {
            matches = *currentMatch;
            int x = std::stoi(matches[1]);
            int y = std::stoi(matches[2]);

            totalSum += x * y;

            ++currentMatch;
        }
    }
    std::cout << "Result: " << totalSum << "\n";

    return 0;
}